airflow/bin/cli.py

```diff
@@ -336,6 +336,34 @@ def list_tasks(args, dag=None):
         print("\n".join(sorted(tasks)))
 
 
+def parsedate_enhanced(args, dag, task):
+    '''
+    In addition to supporting date parsing, this also supports
+    @now,
+    @start_date (i.e. dag start date),
+    @end_date (i.e. dag end date if one exists), and
+    @latest (i.e. run closest to and before now that honors the schedule interval)
+    '''
+    timestr = args.execution_date
+    now = datetime.now()
+    effective_start_date = dag.default_args['start_date'] if (dag.default_args and 'start_date' in dag.default_args) else now
+    effective_end_date = dag.default_args['end_date'] if (dag.default_args and 'end_date' in dag.default_args) else now
+
+    if timestr == '@now':
+        return now
+    elif timestr == '@start_date':
```

Comment:
Rather than this, wonder if `@start_date/@end_date` are set, but not defined for the DAG, should it throw an exception rather than returning `now`?